### PR TITLE
SHO-61: Menu component

### PR DESCRIPTION
**Pull request by saulprl**
## Describe your changes

This PR adds a new component: Menu.

## Task URL (Asana, Jira, etc...)

[SHO-61: Menu](https://cultoperrocafe.nifty.pm/l/npkOW7Jjnslv)

## What was done in this pull request

- [x] Added a new component: `<Menu />`
- [x] Added `onClick` prop to `<HorizontalButton />`
- [x] Storybook with its only variant.
- [x] Added to `index.tsx`
- [x] Made sure that the `Carousel` component is not present in the DOM on desktop layout.

## Demo / Screenshot / Video

[SHO-61-menu.webm](https://user-images.githubusercontent.com/39208827/231969978-45890012-88a6-4eb8-a5aa-4487879848aa.webm)

### CODE REVIEW

File: app/components/atoms/arrow-button/arrow-button.styles.tsx [hunk 1]
```diff
@@ -16,8 +16,8 @@ const arrowTransform = {
 
 export const ArrowButtonFace = styled.span`
   display: block;
-  width: 100%;
-  height: 100%;
+  width: 46px;
+  height: 46px;
```
Comment (tonymtz) [inline app/components/atoms/arrow-button/arrow-button.styles.tsx:20]: image

File: app/components/atoms/arrow-button/arrow-button.styles.tsx [hunk 2]
```diff
@@ -69,4 +69,8 @@ export const StyledArrowButton = styled.button<ArrowButtonProps>`
     filter: brightness(0.95);
     transform: translateY(-28px);
   }
+
+  * {
+    box-sizing: border-box;
+  }
```
Comment (tonymtz) [inline app/components/atoms/arrow-button/arrow-button.styles.tsx]: Could you explain why this is needed?

Comment (saulprl) [reply app/components/atoms/arrow-button/arrow-button.styles.tsx]: When I was working on the NavBar I noticed that it wasn't taking all of the available width, so a solution was to set box-sizing to border-box, which is actually what I generally work with. However, `global.styles` don't apply to the Storybook components, so I added border-box to those components in order to make them look like they're supposed to. That's also why I had to change the width/height values.

Comment (tonymtz) [reply app/components/atoms/arrow-button/arrow-button.styles.tsx]: I agree with having it global. But I don't with having it in every component: DRY.
So let's make it global then!

File: .storybook/preview.js [hunk 3]
```diff
@@ -16,6 +15,7 @@ export const parameters = {
 export const decorators = [
   (Story) => (
     <ThemeProvider theme={theme}>
+      <GlobalStyles />
```
Comment (tonymtz) [inline .storybook/preview.js:18]: 👌🏻

**Review by tonymtz — APPROVED** [review]

**Pull request merged**